clang/lib/Sema/CheckExprLifetime.cpp

```diff
@@ -439,6 +378,8 @@ static void visitLifetimeBoundArguments(IndirectLocalPath &Path, Expr *Call,
   if (!Callee)
     return;
 
+  bool EnableGSLAnalysis = !Callee->getASTContext().getDiagnostics().isIgnored(
```

Comment:
Can we move this to `VisitGSLPointerArg` and do all the calls to it unconditionally ?
This would remove the 3–4 usages in `if` conditions as well.

FWIW, I don't think this would effect performance with a few extra calls, as it only exists for optionally turning this analysis off.

Reply:
> Can we move this to VisitGSLPointerArg and do all the calls to it unconditionally ?

Yeah, we can do that. The code is a bit tricky now, and I think keeping the `if (EnableGSLAnalysis)` condition make it easier to understand and reason about  (explicitly indicating we're handling the GSL case).